src/api/cluster_create.go

```diff
@@ -110,9 +110,41 @@ func (a *ClusterAPI) CreateCluster(c *gin.Context) {
 			createClusterRequest.SecretId = secret.GenerateSecretIDFromName(createClusterRequest.SecretName)
 		}
 
-		commonCluster, err := a.createCluster(ctx, &createClusterRequest, orgID, userID, createClusterRequest.PostHooks)
-		if err != nil {
-			c.JSON(err.Code, err)
+		existingCluster, _ := a.clusterManager.GetClusterByName(ctx, orgID, createClusterRequest.Name)
```

Comment:
**Question**: wouldn't it be better if we check the returned error here instead of assuming `if existingCluster == nil then GetClusterByName() returned nil, NotFoundError{}`?
I see some benefit and safety if we handled only the `NotFoundError` (or whatever is returned in that case) as not existing and proceeding and for every other error I would immediately return 500[1], but if the cluster is successfully retrieved then I would do the following if branch's logic.

Does it make sense what I'm trying to say? What do you think?

[1]: This is the case which is handled as not existing cluster right now and the cluster creation is tried again which of course is going to fail if there is a cluster we just couldn't retrieve it this time by name.

Reply:
Unfortunately I could only check the error message, not the type of the error it is bundled up with other error messages, but it works!